src/Polly.Core/Strategy/ResilienceStrategyTelemetry.cs

```diff
@@ -29,7 +29,7 @@ internal ResilienceStrategyTelemetry(ResilienceTelemetrySource source, Diagnosti
     public void Report<TArgs>(string eventName, TArgs args)
         where TArgs : IResilienceArguments
     {
-        if (!DiagnosticSource.IsEnabled(eventName))
+        if (DiagnosticSource == null || !DiagnosticSource.IsEnabled(eventName))
```

Comment:
```suggestion
        if (DiagnosticSource is null || !DiagnosticSource.IsEnabled(eventName))
```

Reply:
Yeah, I need to remember to use this syntax :)